t/05-distribution.t

```diff
@@ -61,6 +61,56 @@ subtest 'script_run' => sub {
     is $cmdcall->{buffer_size}, 6272, 'appropriate buffer size used for long command';
 };
 
+subtest 'set expected serial and autoinst failures' => sub {
+
+    my $d = distribution->new;
+    $d->set_expected_serial_failures([
+            {type => 'soft', message => 'Soft Serial Failure Message 1', pattern => qr/Test Pattern1/},
+            {type => 'soft', message => 'Soft Serial Failure Message 2', pattern => qr/Test Pattern2/},
+            {type => 'hard', message => 'Hard Serial Failure Message 3', pattern => qr/Test Pattern3/},]
+    );
+    is_deeply($d->{serial_failures}, [
+            {
+                type => 'soft',
+                pattern => qr/Test Pattern1/u,
+                message => 'Soft Serial Failure Message 1'
+            },
+            {
+                type => 'soft',
+                message => 'Soft Serial Failure Message 2',
+                pattern => qr/Test Pattern2/u
+            },
+            {
+                type => 'hard',
+                message => 'Hard Serial Failure Message 3',
+                pattern => qr/Test Pattern3/u
+            }
+    ], 'Expected serial_failures matched');
+
+    $d->set_expected_autoinst_failures([
+            {type => 'soft', message => 'Soft autoinst Failure Message 1', pattern => qr/Test Pattern1/},
+            {type => 'soft', message => 'Soft autoinst Failure Message 2', pattern => qr/Test Pattern2/},
+            {type => 'hard', message => 'Hard autoinst Failure Message 3', pattern => qr/Test Pattern3/},]
+    );
+    is_deeply($d->{autoinst_failures}, [
+            {
+                type => 'soft',
+                pattern => qr/Test Pattern1/u,
+                message => 'Soft autoinst Failure Message 1'
```

Comment:
Duplicate of line 74 with only 1 change, is it possible to create a generic subroutine?

Reply:
@Wabri Please re-review. I have attempted to remove duplicate lines.